### PR TITLE
Add WavPack audio files support

### PR DESCRIPTION
**Pull request by lucianoiam**
[WavPack](https://github.com/dbry/WavPack) is a lossless audio codec

### CODE REVIEW

File: external/st_audiofile/src/st_audiofile.c [hunk 1]
```diff
@@ -449,6 +496,25 @@ uint64_t st_read_s16(st_audio_file* af, int16_t* buffer, uint64_t count)
     case st_audio_file_mp3:
         count = drmp3_read_pcm_frames_s16(af->mp3, count, buffer);
         break;
+    case st_audio_file_wv:
+        {
+            uint32_t channels = af->cache.wv.channels;
+            int32_t* buf_i32 = (int32_t*)malloc(4 * channels * count);
```
Comment (paulfd) [inline external/st_audiofile/src/st_audiofile.c:504]: May I change this for some C++-y lifetime management version?

Comment (lucianoiam) [reply external/st_audiofile/src/st_audiofile.c:504]: Yes sure, that would be great.

Comment (paulfd) [reply external/st_audiofile/src/st_audiofile.c:504]: Nevermind I forgot this is a C file.

File: external/st_audiofile/src/st_audiofile.h [hunk 2]
```diff
@@ -19,6 +19,8 @@
 #include <wchar.h>
 #endif
 
+#define WAVPACK_MEMORY_ASSUMED_VERSION 5
```
Comment (paulfd) [inline external/st_audiofile/src/st_audiofile.h]: Does this need to be in the header file?

File: external/st_audiofile/src/st_audiofile.c [hunk 3]
```diff
@@ -478,6 +544,28 @@ uint64_t st_read_f32(st_audio_file* af, float* buffer, uint64_t count)
...
+            if (!buf_i32) {
+                return 0;
+            }
+            count = channels * WavpackUnpackSamples(af->wv, buf_i32, (uint32_t)count);
+            if (!(af->cache.wv.mode & MODE_FLOAT)) {
+                if (af->cache.wv.bitrate < 32) {
+                    int d = 32 - af->cache.wv.bitrate;
+                    for (uint64_t i = 0; i < count; i++) {
+                        buf_i32[i] <<= d;
```
Comment (paulfd) [inline external/st_audiofile/src/st_audiofile.c:563]: So basically in wavpack if you use integers less than 32 bit length, you're not gaining any space?

**Review by paulfd — APPROVED** [review]
I rebased and added a test (and force-pushed on your branch).

File: external/st_audiofile/src/st_audiofile.h (hunk 2 above)
Comment (paulfd) [reply external/st_audiofile/src/st_audiofile.h]: I moved it for now, will revert if it becomes an issue.

Comment (lucianoiam) [reply external/st_audiofile/src/st_audiofile.h]: No reason to have it in the header file, https://github.com/sfztools/sfizz/pull/1129/commits/c3c416496f26b41b27c82d8e1adf750e4e24ed27 looks good.

File: external/st_audiofile/src/st_audiofile.c (hunk 3 above)
Comment (lucianoiam) [reply external/st_audiofile/src/st_audiofile.c:563]: Acoording to a [comment](https://github.com/dbry/WavPack/blob/ffc72be679dcb13b7e6004060c4e7baf5ba24401/src/unpack_utils.c#L94) in `WavpackUnpackSamples()`, the function always returns 32-bit data, padded when needed. So optimizing sfizz to use less memory when dealing with < 32 bit WV audio does not seem possible.

Comment (paulfd) [reply external/st_audiofile/src/st_audiofile.c:563]: OK np, in memory we store as float anyway. It was just out of curiosity.

**Pull request merged**